waspc/src/Wasp/Generator/ServerGenerator/OAuthAuthG.hs

```diff
@@ -0,0 +1,118 @@
+module Wasp.Generator.ServerGenerator.OAuthAuthG
+  ( genOAuthAuth,
+    depsRequiredByPassport,
+  )
+where
+
+import Data.Aeson (object, (.=))
+import qualified Data.Aeson as Aeson
+import Data.Maybe (fromJust, isJust)
+import StrongPath
+  ( Dir,
+    File',
+    Path,
+    Path',
+    Posix,
+    Rel,
+    reldir,
+    reldirP,
+    relfile,
+    (</>),
+  )
+import qualified StrongPath as SP
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec.App as AS.App
+import qualified Wasp.AppSpec.App.Auth as AS.App.Auth
+import qualified Wasp.AppSpec.App.Auth as AS.Auth
+import qualified Wasp.AppSpec.App.Dependency as App.Dependency
+import Wasp.AppSpec.Valid (getApp)
+import Wasp.Generator.AuthProviders (gitHubAuthInfo, googleAuthInfo)
+import Wasp.Generator.AuthProviders.OAuth (OAuthAuthInfo)
+import qualified Wasp.Generator.AuthProviders.OAuth as OAuth
+import Wasp.Generator.FileDraft (FileDraft)
+import Wasp.Generator.Monad (Generator)
+import Wasp.Generator.ServerGenerator.Common (ServerSrcDir)
+import qualified Wasp.Generator.ServerGenerator.Common as C
+import Wasp.Generator.ServerGenerator.JsImport (extImportToImportJson)
+import Wasp.Util ((<++>))
+
+genOAuthAuth :: AS.Auth.Auth -> Generator [FileDraft]
+genOAuthAuth auth
+  | AS.Auth.isExternalAuthEnabled auth =
+      genOAuthHelpers
+        <++> genOAuthProvider googleAuthInfo (AS.Auth.google . AS.Auth.methods $ auth)
+        <++> genOAuthProvider gitHubAuthInfo (AS.Auth.gitHub . AS.Auth.methods $ auth)
+  | otherwise = return []
+
+genOAuthHelpers :: Generator [FileDraft]
+genOAuthHelpers =
+  sequence
+    [ return $ C.mkSrcTmplFd [relfile|auth/providers/oauth/init.ts|],
+      return $ C.mkSrcTmplFd [relfile|auth/providers/oauth/setupRouter.ts|]
+    ]
+
+genOAuthProvider ::
+  OAuthAuthInfo ->
+  Maybe AS.Auth.ExternalAuthConfig ->
+  Generator [FileDraft]
+genOAuthProvider authInfo maybeUserConfig
+  | isJust maybeUserConfig =
+      sequence
+        [ genOAuthConfig authInfo maybeUserConfig $ [reldir|auth/providers/config|] </> providerTsFile
+        ]
+  | otherwise = return []
+  where
+    providerTsFile :: Path' (Rel ()) File'
+    providerTsFile = fromJust $ SP.parseRelFile $ slug ++ ".ts"
+
+    slug = OAuth.slug authInfo
+
+genOAuthConfig ::
```

Comment:
Used to generate the specific provider config based on the generic `oauth.ts` file. The config receives everything: auth info, npm packages, user defined imports and env variables. It's all in one config file.